### PR TITLE
Implemented repository services and updated database for episodes feature

### PR DESCRIPTION
**Pull request by Faka-GLB**
Tasks completed:
- Created service call to get episodes from API.
- Updated Room database adding episode entity and Dao with getAllEpisodes() and insert operations.
- Added episode classes to mapper.
- Updated Gradle to 7.0.3

|Data module| Domain module |
|---------------|------------------|
|rick and morty data module | rick and morty domain module |

In-app data preview (screen in progress):
image

Test Results:

| UseCase | ServiceMapper | RoomMapper|
|----------|-------------------|---------------|
|uc test | mapper test |room test |

In progress:
- Viewmodel and fragment
- episode item
- motion animation
- RecyclerAdapter
- Values
- UT

### CODE REVIEW

**Review by BQuinteros — APPROVED** [review]

File: data/src/main/java/com/onboarding/data/service/RickAndMortyRepositoryImpl.kt [hunk 1]
```diff
@@ -3,24 +3,33 @@ package com.onboarding.data.service
...
             Result.Success((response.body() as CharacterBaseResponse).transform())
         } catch (e: Exception) {
             Result.Failure(Exception(EXCEPTION_MESSAGE))
         }
     }
 
+    override fun getAllEpisodes(): Result<EpisodeBase> {
+        return try {
+            val response = callResponse.callAllEpisodes().execute()
+            Result.Success((response.body() as EpisodeBaseResponse).transform())
```
Comment (SartiMau) [inline data/src/main/java/com/onboarding/data/service/RickAndMortyRepositoryImpl.kt]: here you should check if the response is successful before setting the result. Same in the getAllCharacters method

File: di/src/main/java/com/onboarding/di/koinModule.kt [hunk 2]
```diff
@@ -24,10 +27,13 @@ val repositoryModule = module {
             androidContext(),
             RickAndMortyRoomDatabase::class.java,
             ConstantUtil.DATABASE_NAME
-        ).build()
+        ).fallbackToDestructiveMigration().build()
```
Comment (SartiMau) [inline di/src/main/java/com/onboarding/di/koinModule.kt]: what does the fallbackToDestructiveMigration invocation?

**Review by SartiMau — CHANGES_REQUESTED** [review]

File: di/src/main/java/com/onboarding/di/koinModule.kt (hunk 2 above)
Comment (Faka-GLB) [reply di/src/main/java/com/onboarding/di/koinModule.kt]: This recreate the database files and schema when a migration occurs by deleting every table if a migration file is not provided.

File: data/src/main/java/com/onboarding/data/service/RickAndMortyRepositoryImpl.kt (hunk 1 above)
Comment (Faka-GLB) [reply data/src/main/java/com/onboarding/data/service/RickAndMortyRepositoryImpl.kt]: done

File: di/src/main/java/com/onboarding/di/koinModule.kt (hunk 2 above)
Comment (SartiMau) [reply di/src/main/java/com/onboarding/di/koinModule.kt]: As we just talked, please do the migration instead of call this method

Comment (Faka-GLB) [reply di/src/main/java/com/onboarding/di/koinModule.kt]: done

**Review by SartiMau — APPROVED** [review]

File: data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt [hunk 3]
```diff
@@ -40,3 +44,18 @@ private fun LocationLinkResponse.transform() = LocationLink(
     this.locationName,
     this.locationUrl
 )
+
+fun EpisodeBaseResponse.transform() = EpisodeBase(
+    this.info.transform(),
+    this.results.mapTo(mutableListOf(), { it.transform() })
+)
+
+private fun EpisodeResultResponse.transform() = EpisodeResult(
```
Comment (nicolbidone) [inline data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt]: please use a more explicit names like transformToEpisodeResult

File: data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt [hunk 4]
```diff
@@ -40,3 +44,18 @@ private fun LocationLinkResponse.transform() = LocationLink(
     this.locationName,
     this.locationUrl
 )
+
+fun EpisodeBaseResponse.transform() = EpisodeBase(
```
Comment (nicolbidone) [inline data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt]: please use a more explicit names like transformToEpisodeBase

File: data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt (hunk 3 above)
Comment (Faka-GLB) [reply data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt]: renamed

File: data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt (hunk 4 above)
Comment (Faka-GLB) [reply data/src/main/java/com/onboarding/data/mapper/ServiceMapper.kt]: renamed

**Review by nicolbidone — APPROVED** [review]

**Pull request merged**